development/image-syncer/main.go

```diff
@@ -0,0 +1,228 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/jamiealquiza/envy"
+	parser "github.com/novln/docker-parser"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	log = logrus.New()
+)
+
+type SyncDef struct {
+	TargetRepoPrefix string `yaml:"targetRepoPrefix"`
+	Images           []Image
+}
+
+type Image struct {
+	Source string
+}
+
+type Config struct {
+	ImagesFile    string
+	TargetKeyFile string
+	DryRun        bool
+}
+
+func getAuthString(user, password string) (string, error) {
+	authConfig := types.AuthConfig{
+		Username: user,
+		Password: password,
+	}
+	encodedJSON, err := json.Marshal(authConfig)
+	if err != nil {
+		return "", err
+	}
+	return base64.URLEncoding.EncodeToString(encodedJSON), nil
+}
+
+func getImageIdAndRepoDigest(ctx context.Context, cli *client.Client, authString, image string) (string, string, error) {
```

Comment:
The `authString` does not seem to be used in the function. Will it have a purpose in a future?

Reply:
I tested registries where it was used, but since we don't have a situation like that I will remove that

Reply:
fixed